### PR TITLE
The Swahili Classification Task

### PR DESCRIPTION
**Pull request by msamwelmollel**
## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding datasets checklist

**Reason for dataset addition**: ... 

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding a model checklist

 - [ ] I have filled out the ModelMeta object to the extent possible
 - [ ] I have ensured that my model can be loaded using
   - [ ] `mteb.get_model(model_name, revision_id)` and
   - [ ] `mteb.get_model_meta(model_name, revision_id)`
 - [x] I have tested the implementation works on a representative set of tasks.

### CODE REVIEW

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# from ....abstasks import AbsTaskClassification
+from mteb.abstasks.AbsTaskClassification import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SwahiliNewsClassification(AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="SwahiliNewsClassification",
+        description="Dataset for Swahili News Classification, categorized with 5 domains. Building and Optimizing Swahili Language Models: Techniques, Embeddings, and Datasets",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/swa/SwahiliNewsClassification.py:11]: which 5 domains?

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        date=("2019-01-01", "2023-05-01"),
+        form=["written"],
+        dialect=[],
+        domains=["News"],
+        task_subtypes=[],
+        license="CC BY-NC-SA 4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        text_creation="found",
+        bibtex_citation="""
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/swa/SwahiliNewsClassification.py:31]: No citation?

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+# from ....abstasks import AbsTaskClassification
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/swa/SwahiliNewsClassification.py:3]:
```suggestion
```

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from mteb.abstasks.AbsTaskClassification import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SwahiliNewsClassification(AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="SwahiliNewsClassification",
+        description="Dataset for Swahili News Classification, categorized with 5 domains. Building and Optimizing Swahili Language Models: Techniques, Embeddings, and Datasets",
+        reference="https://huggingface.co/datasets/Mollel/SwahiliNewsClassification",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/swa/SwahiliNewsClassification.py:12]: Seems like the dataset has quite limited documentation?

**Review by KennethEnevoldsen — COMMENTED** [review]
Please also specify "reason for adding the dataset".

**Comment by KennethEnevoldsen** [comment]
@msamwelmollel just a friendly reminder that this PR is still open

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py (hunk 1 above)
Comment (msamwelmollel) [reply mteb/tasks/Classification/swa/SwahiliNewsClassification.py:11]: Reason for Adding the Dataset and domain of the classification:

Swahili is spoken by 100-150 million people across East Africa. In Tanzania, for example, it is one of the primary national languages and the official language of instruction in all schools. News in Swahili is an integral part of Tanzania's media sphere.
News contributes to education, technology, and a country's economic growth, and news in local languages plays an important cultural role in many African countries. In the modern age, however, African languages in news and other spheres are at risk of being lost as English becomes the dominant language in online spaces.
The Swahili news dataset was created to bridge the gap in utilizing the Swahili language to create NLP technologies. It aims to assist AI practitioners in Tanzania and across Africa in honing their NLP skills to address various challenges within organizations or societies related to the Swahili language. The Swahili news dataset, sourced from multiple websites providing news in Swahili, is a valuable resource for NLP research and development.
The dataset was curated explicitly for text classification tasks, categorizing news content into six distinct topics. This categorization facilitates the development of robust NLP models that can more effectively understand and process Swahili text. 
Six Domains:
Local News (Kitaifa):
News concerning local events, politics, and developments within Tanzania.
International News (Kimataifa):
Coverage of global events and news stories affecting the international community.
Finance News (Uchumi):
Financial and economic news, including market trends, monetary policies, and business updates.
Health News (Afya):
Information and updates on health-related topics, medical research, public health issues, and wellness.
Sports News (Michezo):
News related to sports events, athletes, competitions, and sports culture locally and internationally.
Entertainment News (Burudani):
Coverage of entertainment industry news, including celebrity updates, music, movies, and cultural events.
By integrating this dataset into MTEB, we aim to support the development of NLP models capable of understanding and processing Swahili text across these vital domains, thus promoting linguistic diversity and technological advancement in East Africa. The high accuracy of this dataset is attributed to the human annotation process involved in its creation.

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py (hunk 2 above)
Comment (msamwelmollel) [reply mteb/tasks/Classification/swa/SwahiliNewsClassification.py:31]: @inproceedings{davis2020swahili,
        title = "Swahili: News Classification Dataset (0.2)",
        author = "Davis, David",
        year = "2020",
        publisher = "Zenodo",
        doi = "10.5281/zenodo.5514203",
        url = "https://doi.org/10.5281/zenodo.5514203"
        }
        """,

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py (hunk 4 above)
Comment (msamwelmollel) [reply mteb/tasks/Classification/swa/SwahiliNewsClassification.py:12]: I have added to the HF the documentation of dataset

**Comment by msamwelmollel** [comment]
> @msamwelmollel just a friendly reminder that this PR is still open

Thank you for the reminder. I have worked on your recommendations and added more information based on your review. I would like to know if I need to push the changes again, or if you are going to resolve them. Additionally, there are a few minor adjustments I need to make in the pull request based on your feedback.

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/swa/SwahiliNewsClassification.py:31]: Wonderful. Feel free to add that in as well.

File: mteb/tasks/Classification/swa/SwahiliNewsClassification.py (hunk 4 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/swa/SwahiliNewsClassification.py:12]: Thanks @msamwelmollel much appreciated

**Comment by KennethEnevoldsen** [comment]
@msamwelmollel it looks like all the recommendations are correct, do add them to the PR, you might also need to pull from main to ensure that all tests pass (this is due to a recent update in a downstream package causing breaking changes).

**Comment by msamwelmollel** [comment]
> @msamwelmollel it looks like all the recommendations are correct, do add them to the PR, you might also need to pull from main to ensure that all tests pass (this is due to a recent update in a downstream package causing breaking changes).

Done @KennethEnevoldsen

**Review by KennethEnevoldsen — APPROVED** [review]

**Comment by KennethEnevoldsen** [comment]
Perfect enable auto-merge (it will merge assuming all tests pass)

**Pull request merged**